server/py/framework/db/sqldb/db.py

```diff
@@ -525,10 +526,17 @@ def del_runs(
     def _enrich_run_struct_from_model(
         self, run: Run, run_struct: dict, with_notifications: bool
     ):
+        status = run_struct.setdefault("status", {})
+
+        # These fields are saved in struct as timestamps with fsp=6, while the corresponding columns
+        # in the database have fsp=3. Since 'ORDER BY' is applied to the column, we return the value from
+        # the column (not from the struct) to ensure the ordering is correct.
+        # In SQLite, the start_time and updated columns return timestamps with fsp=6.
+        status["start_time"] = self._add_utc_timezone(run.start_time).isoformat()
+        status["last_update"] = self._add_utc_timezone(run.updated).isoformat()
```

Comment:
if run.start_time / run.updated is None, it will raise an exception.
I suggest to condition that on whether it is set
```suggestion
for status_field, struct_field in [("end_time", "end_time"), ("start_time", "start_time"), ("last_update", "update")]:
        if field_value := getattr(run, struct_field, None):
          status[status_field] = self._add_utc_timezone(field_value).isoformat()
```